utils/vpr_pbtype_arch_wrapper.py

```diff
@@ -0,0 +1,404 @@
+        outfile,
+    )
+
+    # Pin locations
+    ploc = ET.SubElement(
+        tile,
+        "pinlocations",
+        {"pattern": "custom"},
+    )
+    ilocs = []
+    olocs = []
+    for i in range(0, tile_height):
+        ilocs.append(
+            ET.SubElement(
+                ploc,
+                "loc",
+                {
+                    "side": "left",
+                    "xoffset": "0",
+                    "yoffset": str(i)
+                },
+            )
+        )
+        ilocs[i].text = ""
+        olocs.append(
+            ET.SubElement(
+                ploc,
+                "loc",
+                {
+                    "side": "right",
+                    "xoffset": "0",
+                    "yoffset": str(i)
+                },
+            )
+        )
+        olocs[i].text = ""
+
+    # Interconnect
+    connect = ET.SubElement(
+        tile,
+        "interconnect",
+    )
+
+    # Clock pins
+    for s, d in flatten(clocks):
+        input_tag = ET.SubElement(
+            tile,
+            "clock",
+            {
+                "name": s,
+                "num_pins": "1",
+                "equivalent": "none"
+            },
+        )
+        ET.SubElement(
+            connect,
+            "direct",
+            {
+                "input": "TILE.{}".format(s),
+                "name": "TILE.{}-{}.{}".format(s, name, d),
+                "output": "{}.{}".format(name, d),
+            },
+        )
+
+        for i in range(0, tile_height):
+            ilocs[i].text += "TILE.{} ".format(s)
+
+    # Input Pins
+    for s, d in flatten(inputs):
+        input_tag = ET.SubElement(
+            tile,
+            "input",
+            {
+                "name": s,
+                "num_pins": "1",
+                "equivalent": "none"
+            },
+        )
+        ET.SubElement(
+            connect,
+            "direct",
+            {
+                "input": "TILE.{}".format(s),
+                "name": "TILE.{}-{}.{}".format(s, name, d),
+                "output": "{}.{}".format(name, d),
+            },
+        )
+
+        for i in range(0, tile_height):
+            ilocs[i].text += "TILE.{} ".format(s)
+
+    # Output Pins
+    for d, s in flatten(outputs):
+        output_tag = ET.SubElement(
+            tile,
+            "output",
+            {
+                "name": d,
+                "num_pins": "1",
+                "equivalent": "none"
+            },
+        )
+        ET.SubElement(
+            connect,
+            "direct",
+            {
+                "input": "{}.{}".format(name, s),
+                "name": "TILE.{}-{}.{}".format(s, name, d),
+                "output": "TILE.{}".format(d),
+            },
+        )
+        for i in range(0, tile_height):
+            olocs[i].text += "TILE.{} ".format(d)
+
+    return name
+
+
+def pretty_xml(xml):
+    """Use xmllint to prettify the XML output."""
```

Comment:
Worth mentioning that the prettify XML output is returned as a string?

Reply:
Done.

Reply:
Done.